### DISCUSSION TITLE
How to use Jenkins declarative pipeline for releasing?

### DISCUSSION THREAD

**Discussion by poloka** [lerna/lerna#3710] [Q&A]
I have a simple declarative pipeline that is to run a `lerna publish` command on a monorepo that has two packages.  Versioning is defined as independent.  Has anyone found a way to either 1) capture the lerna output that requests version updates to pass back to the CI for input? or 2) found a way to use jenkins pipeline input as the 'custom' version that you can then detect the lerna prompt and then select the custom version and past in the prompted version?

**Comment by fahslaj** [comment]
Hi @poloka , you can pass in a specific version to `lerna version` or `lerna publish`. For example:

`lerna publish 3.0.2`

If you can pass an input version into your CI job, then you could use that input variable in this command to tell Lerna to use that specific version:

`lerna publish ${MY_CUSTOM_VERION}`

**Comment by melygosh** [comment]
Hi there 👋 
I'm trying to do smth like this at the moment and curious whether some guideline is available for Lerna & Jenkins?

We have a component library that I move to Lerna. For CI/CD we use Jenkins.
What I'm trying to achieve is the flow where changes pushed/merged to particular branch (i.e. `main`) should trigger "prerelease" scenario and each affected package published with next version like 1.1.0-alpha.1 -> 1.1.0-alpha.2. 

Currently it looks like I do smth wrong:
```
stage('Version & Publish') {
    // only run this stage if commit message does not contain a release commit
    when { not { changelog "^chore\\(release\\):.*" } }
    steps {
        script {
            if (isDevelop(env.BRANCH_NAME)) {
                // 1.1.0-alpha.1 -> 1.1.0-alpha.2
                execNpm 'run version:prerelease'
                // sync the develop branch with the remote repository
                execute 'git push origin develop'
                // publish 
                execNpm 'run release'
            }
        }
    }
}
```

version:prerelease script looks like this: `lerna version prerelease --preid alpha --yes`
release script `lerna publish from-git --dist-tag stable --pre-dist-tag latest --yes`

Is there any possible way to improve or refactor the logic? 
I've been trying to find anything useful for a last couple of days but no success so far.
